src/sknnr_spatial/image/dataarray.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import dask.array as da
+import numpy as np
+import xarray as xr
+
+from ._base import ImagePreprocessor, kneighbors, predict
+from ._dask_backed import (
+    kneighbors_from_dask_backed_array,
+    predict_from_dask_backed_array,
+)
+
+if TYPE_CHECKING:
+    from numpy.typing import NDArray
+    from sklearn.base import BaseEstimator
+    from sklearn.neighbors import KNeighborsRegressor
+
+
+class DataArrayPreprocessor(ImagePreprocessor):
+    """
+    Pre-processor for multi-band xr.DataArrays.
+    """
+
+    _backend = da
+    band_dim = 0
+
+    def _validate_nodata_vals(
+        self, nodata_vals: float | tuple[float] | NDArray | None
+    ) -> NDArray | None:
+        """
+        Get an array of NoData values in the shape (bands,) based on user input and
+        DataArray metadata.
+        """
+        # Defer to user-provided nodata values over stored attributes
+        if nodata_vals is not None:
+            return super()._validate_nodata_vals(nodata_vals)
+
+        # If present, broadcast the _FillValue attribute to all bands
+        fill_val = self.image.attrs.get("_FillValue")
+        if fill_val is not None:
+            return np.full((self.n_bands,), fill_val)
+
+        return None
+
+    def _flatten(self, image: xr.DataArray) -> xr.DataArray:
+        """Flatten the dataarray from (bands, y, x) to (pixels, bands)."""
```

Comment:
Minor terminology.  I think elsewhere you've been using `samples` instead of `pixels`.  Just wanted to flag this in case you wanted to review.

Reply:
I think I have been using samples when discussing the design, but it looks like I used pixels throughout the comments and docstrings. Any preference between the two? It would be good to settle on one to use consistently, and I'm happy with either.

Reply:
I'm happy with either as well.  Like the "bands" and "channels" discussion, probably settle on one, but perhaps document that both are acceptable.